### PR TITLE
Merge security fixes branch into master

### PR DESCRIPTION
**Pull request by daniel-beck**
This is the branch I used for the last two security updates, fixing three issues:

https://jenkins.io/security/advisory/2018-06-04/#SECURITY-799
https://jenkins.io/security/advisory/2018-06-04/#SECURITY-804
https://jenkins.io/security/advisory/2018-06-25/#SECURITY-915

Not sure how you'd typically handle this, so opening PR to request merge of this branch into master to prevent further diverging histories.

I don't know what's missing to make https://github.com/jenkinsci/github-plugin/releases/tag/v1.29.1 and https://github.com/jenkinsci/github-plugin/releases/tag/v1.29.2 look nice, so I haven't done that.

---
This change is [Reviewable](https://reviewable.io/reviews/jenkinsci/github-plugin/196)

### CODE REVIEW

File: src/main/resources/org/jenkinsci/plugins/github/config/GitHubPluginConfig/config.groovy [hunk 1]
```diff
@@ -29,7 +29,7 @@ f.section(title: descriptor.displayName) {
                             field: "overrideHookUrl",
                             checked: instance.overrideHookURL) {
                         f.entry(field: "hookUrl") {
-                            f.textbox()
+                            f.textbox(checkMethod: "post")
```
Comment (KostyaSha) [inline src/main/resources/org/jenkinsci/plugins/github/config/GitHubPluginConfig/config.groovy]: lol, what is used by default? get?

File: src/test/java/org/jenkinsci/plugins/github/config/GitHubPluginConfigTest_SEC799.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.net.URL;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.not;
+
+//TODO this class can be merged with GitHubPluginConfigTest after the security fix
+public class GitHubPluginConfigTest_SEC799 {
```
Comment (KostyaSha) [inline src/test/java/org/jenkinsci/plugins/github/config/GitHubPluginConfigTest_SEC799.java]: please name correctly classes and methods.

**Comment by KostyaSha** [comment]
> Not sure how you'd typically handle this

Handling what? Making releases without asking maintainers?

File: src/test/java/org/jenkinsci/plugins/github/config/GitHubPluginConfigTest_SEC799.java (hunk 2 above)
Comment (daniel-beck) [reply src/test/java/org/jenkinsci/plugins/github/config/GitHubPluginConfigTest_SEC799.java]: As a matter of policy, we create new classes for tests added in security fixes to make potential backports of fixes easier. As the TODO comment indicates, this is not intended to be a long lived class.

**Comment by daniel-beck** [comment]
> Making releases without asking maintainers?

Both releases were coordinated with @lanwen, and I got explicit approval both for the actual fixes in private code review, as well as for releasing the plugin, so I don't know what you mean. If I should work with you in these cases in the future, that's fine with me -- I just expected @lanwen to tell me if that was the case.

File: src/main/resources/org/jenkinsci/plugins/github/config/GitHubPluginConfig/config.groovy (hunk 1 above)
Comment (daniel-beck) [reply src/main/resources/org/jenkinsci/plugins/github/config/GitHubPluginConfig/config.groovy]: Sadly, yes.

**Comment by KostyaSha** [comment]
> Both releases were coordinated with @lanwen, and I got explicit approval both for the actual fixes in private code review, as well as for releasing the plugin, so I don't know what you mean.

You know where maintainer list is for this plugin and especially you know that i maintain github plugins.

**Comment by lanwen** [comment]
@KostyaSha all ok, thats my approve. I think better to have this fixed in any way as soon as possible. Naming we can handle by ourselves if its needed

**Comment by lanwen** [comment]
@daniel-beck thx for fixes and your time

**Comment by daniel-beck** [comment]
@lanwen You're welcome. Could you confirm that I should contact @KostyaSha instead of you in the future for issues related to this plugin?

**Comment by KostyaSha** [comment]
Not instead, you should contact **both** and for any other plugin in jenkinsci it should be good tone to contact maintainers listed in permissions. My company and production has dependencies on this plugins and you can see that i always had activity in github related plugins in last years.

**Comment by lanwen** [comment]
Yep, now @KostyaSha is first contact. As I can be too laggy to handle requests effectively due to family reasons (I have 9m old child and recently relocated to another country :))

**Comment by KostyaSha** [comment]
> I have 9m old child

Congratulations!

**Comment by daniel-beck** [comment]
@lanwen Thanks, will do. And congrats!

----

@KostyaSha 

> Not instead, you should contact both and for any other plugin in jenkinsci it should be good tone to contact maintainers listed in permissions.

With @lanwen's answer, I see this as being resolved now for this plugin specifically. More generally, as this is the first occurrence of a problem like this in several years of working with maintainers, I'll wait for further instances before adopting a much more time consuming communication process.

**Comment by KostyaSha** [comment]
i merged few changes, but they were minor, so i didn't make release. Now master is mixed and new release is required.

**Comment by jepperson2** [comment]
Any update on the release of 1.29.3? 

Seems like it would address some vulnerabilities we are concerned about.

**Pull request merged**